packages/react/src/auto/AutoFormActionValidators.ts

```diff
@@ -44,19 +45,48 @@ export const validateAutoFormProps = (props: Parameters<typeof useAutoForm>[0])
 };
 
 export const validateTriggersFromMetadata = (metadata?: ActionMetadata | GlobalActionMetadata) => {
-  if (metadata) {
-    // When an action's API trigger is removed but the referenced API client still uses the old api-triggered action, the presence of the trigger shall be checked in the metadata
-
-    const triggersAsArray =
-      (metadata.__typename === "GadgetGlobalAction"
-        ? metadata.triggers
-        : metadata.__typename === "GadgetModel"
-        ? metadata.action.triggers
-        : []) ?? [];
-
-    const hasApiTrigger = triggersAsArray.some((trigger) => trigger.specID === GadgetApiTriggerSpecId);
-    if (!hasApiTrigger) {
-      throw new Error(MissingApiTriggerErrorMessage);
-    }
+  if (!metadata) {
+    return;
+  }
+  // When an action's API trigger is removed but the referenced API client still uses the old api-triggered action, the presence of the trigger shall be checked in the metadata
+
+  const triggersAsArray =
+    (metadata.__typename === "GadgetGlobalAction"
+      ? metadata.triggers
+      : metadata.__typename === "GadgetModel"
+      ? metadata.action.triggers
+      : []) ?? [];
+
+  const hasApiTrigger = triggersAsArray.some((trigger) => trigger.specID === GadgetApiTriggerSpecId);
+  if (!hasApiTrigger) {
+    throw new Error(MissingApiTriggerErrorMessage);
+  }
+};
+
+export const validateFindByObjectWithMetadata = (
+  fields?: readonly { path: string; metadata: FieldMetadata }[],
+  findBy?: RecordIdentifier
+) => {
+  if (!findBy || typeof findBy === "string") {
+    return;
+  }
+
+  if (typeof findBy !== "object" || Array.isArray(findBy) || Object.keys(findBy).length !== 1) {
```

Comment:
we have an `isPlainObject` that might work for this?

Reply:
Good call. Updated to use `isPlainObject` which means there is no need to check `isArray` since that case is prevented with the prototype checks 👍